### PR TITLE
refactor: remove LinqKit

### PR DESCRIPTION
**Pull request by melflitty**
Some functions of LinqKit were used to generate predicate expressions for filters. We don't want to use this package anymore.

### CODE REVIEW

File: Common/src/gRPC/ExpressionExt.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    var rightVisitor = new ReplaceExpressionVisitor(expr2.Parameters[0],
+                                                    parameter);
+    var right = rightVisitor.Visit(expr2.Body);
+
+    return Expression.Lambda<Func<T, bool>>(Expression.OrElse(left,
+                                                              right),
+                                            parameter);
+  }
+
+  private class ReplaceExpressionVisitor : ExpressionVisitor
```
Comment (aneojgurhem) [inline Common/src/gRPC/ExpressionExt.cs:75]: Pretty clean solution !
I'd use Expression.MakeBinary to factorize code and make the generic function private.

File: Common/src/gRPC/ExpressionExt.cs [hunk 2]
```diff
@@ -54,8 +55,8 @@ public static class ExpressionExt
   /// <param name="expr1"> The first predicate expression to combine </param>
   /// <param name="expr2"> The second predicate expression to combine </param>
   /// <returns> A new predicate expression that represents the logical OR of the two expressions </returns>
-  public static Expression<Func<T, bool>> ExpressionOr<T>(this Expression<Func<T, bool>> expr1,
-                                                          Expression<Func<T, bool>>      expr2)
+  public static Expression<Func<T, bool>>? ExpressionOr<T>(this Expression<Func<T, bool>> expr1,
```
Comment (aneojgurhem) [inline Common/src/gRPC/ExpressionExt.cs:58]: should  not return null
(you missed the factorization part)

Comment (melflitty) [reply Common/src/gRPC/ExpressionExt.cs:66]: I was doing it . will push that part after ensuring the generic function works well

Comment (melflitty) [reply Common/src/gRPC/ExpressionExt.cs]: Sorry, I misunderstood your comment . Done!

File: Common/src/gRPC/ExpressionExt.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                                                   parameter);
+    var left = leftVisitor.Visit(expr1.Body);
+
+    var rightVisitor = new ReplaceExpressionVisitor(expr2.Parameters[0],
+                                                    parameter);
+    var right = rightVisitor.Visit(expr2.Body);
+
+    return Expression.Lambda<Func<T, bool>>(Expression.MakeBinary(expressionType,
+                                                                  left!,
+                                                                  right!),
```
Comment (aneojgurhem) [inline Common/src/gRPC/ExpressionExt.cs:41]: I think you should properly check for null here !

Comment (lemaitre-aneo) [reply Common/src/gRPC/ExpressionExt.cs:41]: Personally, I consider `!` a proper null check.

**Pull request merged**